src/components/localtunneldebugger/localtunneldebugger.ts

```diff
@@ -0,0 +1,48 @@
+import * as vscode from 'vscode';
+import { LocalTunnelDebugProvider } from './localtunneldebugprovider.extension';
+import { getLocalTunnelDebugProvider } from '../config/config';
+
+export class LocalTunnelDebugger {
+    private readonly providers = Array.of<LocalTunnelDebugProvider>();
+
+    register(provider: LocalTunnelDebugProvider) {
+        console.log(`Registered local tunnel debugger type ${provider.id}`);
+        this.providers.push(provider);
+    }
+
+    startLocalTunnelDebugProvider(target?: any) {
+        const providerSetting: string | undefined = getLocalTunnelDebugProvider();
+        const providerName: string | undefined = providerSetting ? providerSetting : this.providers.map((p) => p.id).sort()[0];
+        const providerToUse: LocalTunnelDebugProvider | undefined = this.providers.find((p) => p.id === providerName);
+
+        // On success start early.
+        if (this.providers.length && providerToUse) {
```

Comment:
Do you need `this.providers.length`?  If `providerToUse` is set then I think that's all we need.

Reply:
Makes sense!